go.mod

```diff
@@ -19,12 +19,13 @@ require (
 	github.com/cosmos/cosmos-sdk v0.50.10
 	github.com/cosmos/gogoproto v1.7.0
 	github.com/cosmos/ics23/go v0.11.0
+	github.com/cosmos/solidity-ibc-eureka/abigen v0.0.0-20241124101149-3e2c74b736f4
```

Comment:
this is a separate go module, right? 

maybe in future could be nice to have all encoding and protobuf stuff in the ibc repo, but that's probably quite far off right now

Reply:
Yes, it is generated code from the ABI definitions of the solidity EVM contracts (Solidity -> ABI -> go).

Reply:
I actually think that we should have the encoding/decoding logic here (or ibc repo I guess). I'd even rather use manually written code to abi encode/decode the packet data than to use autogenerated code from another repo. But for now, this is probably ok...

Reply:
There is a bigger discussion to be had on centralizing types and encoding. For now, this is at least relatively convenient and gives a direct link to where the correct definitions are.